File: app/src/main/java/com/example/thegreatmenu/MainActivity.java
package com.example.thegreatmenu;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainActivity extends BaseActivity {

    private FirebaseAuth mAuth;
    private EditText etEmail, etPassword;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // אתחול Firebase
        mAuth = FirebaseAuth.getInstance();

        // בדיקה אם המשתמש כבר מחובר
        if (mAuth.getCurrentUser() != null) {
            navigateToFeed();
        }

        etEmail = findViewById(R.id.editTextTextEmailAddress);
        etPassword = findViewById(R.id.editTextTextPassword);
        Button btnLogin = findViewById(R.id.btnLogin);
        Button btnRegister = findViewById(R.id.btnRegister);

        // כפתור התחברות
        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = etEmail.getText().toString();
                String password = etPassword.getText().toString();

                if (!email.isEmpty() && !password.isEmpty()) {
                    loginUser(email, password);
                } else {
                    Toast.makeText(MainActivity.this, "נא להזין אימייל וסיסמה", Toast.LENGTH_SHORT).show();
                }
            }
        });

        // כפתור הרשמה
        btnRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = etEmail.getText().toString();
                String password = etPassword.getText().toString();

                if (!email.isEmpty() && !password.isEmpty()) {
                    registerUser(email, password);
                } else {
                    Toast.makeText(MainActivity.this, "נא להזין אימייל וסיסמה להרשמה", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    private void loginUser(String email, String password) {
        mAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            navigateToFeed();
                        } else {
                            Toast.makeText(MainActivity.this, "התחברות נכשלה: " + task.getException().getMessage(),
                                    Toast.LENGTH_SHORT).show();
                        }
                    }
                });
    }

    private void registerUser(String email, String password) {
        mAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            Toast.makeText(MainActivity.this, "נרשמת בהצלחה!", Toast.LENGTH_SHORT).show();
                            navigateToFeed();
                        } else {
                            Toast.makeText(MainActivity.this, "הרשמה נכשלה: " + task.getException().getMessage(),
                                    Toast.LENGTH_SHORT).show();
                        }
                    }
                });
    }

    private void navigateToFeed() {
        Intent intent = new Intent(MainActivity.this, FeedActivity.class);
        startActivity(intent);
        finish(); // מונע חזרה למסך הלוגין
    }
}

File: app/src/main/java/com/example/thegreatmenu/MainActivity.kt
package com.example.thegreatmenu

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {

    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // אתחול Firebase Auth
        auth = FirebaseAuth.getInstance()

        // בדיקה אם המשתמש כבר מחובר - אם כן, דלג ישר לפיד
        if (auth.currentUser != null) {
            navigateToFeed()
        }

        val etEmail = findViewById<EditText>(R.id.editTextTextEmailAddress)
        val etPassword = findViewById<EditText>(R.id.editTextTextPassword)
        val btnLogin = findViewById<Button>(R.id.btnLogin)
        val btnRegister = findViewById<Button>(R.id.btnRegister)

        // לוגיקת התחברות (Login)
        btnLogin.setOnClickListener {
            val email = etEmail.text.toString()
            val password = etPassword.text.toString()

            if (email.isNotEmpty() && password.isNotEmpty()) {
                auth.signInWithEmailAndPassword(email, password)
                    .addOnCompleteListener(this) { task ->
                        if (task.isSuccessful) {
                            navigateToFeed()
                        } else {
                            Toast.makeText(baseContext, "התחברות נכשלה: ${task.exception?.message}",
                                Toast.LENGTH_SHORT).show()
                        }
                    }
            } else {
                Toast.makeText(this, "נא להזין אימייל וסיסמה", Toast.LENGTH_SHORT).show()
            }
        }

        // לוגיקת הרשמה (Register) - ליצירת משתמש מהירה
        btnRegister.setOnClickListener {
            val email = etEmail.text.toString()
            val password = etPassword.text.toString()

            if (email.isNotEmpty() && password.isNotEmpty()) {
                auth.createUserWithEmailAndPassword(email, password)
                    .addOnCompleteListener(this) { task ->
                        if (task.isSuccessful) {
                            Toast.makeText(baseContext, "נרשמת בהצלחה!", Toast.LENGTH_SHORT).show()
                            navigateToFeed()
                        } else {
                            Toast.makeText(baseContext, "הרשמה נכשלה: ${task.exception?.message}",
                                Toast.LENGTH_SHORT).show()
                        }
                    }
            } else {
                Toast.makeText(this, "נא להזין אימייל וסיסמה להרשמה", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun navigateToFeed() {
        val intent = Intent(this, FeedActivity::class.java)
        startActivity(intent)
        finish() // מונע חזרה למסך הלוגין בלחיצה על Back
    }
}